src/server/utils/cdnCache.js

```diff
@@ -89,15 +90,21 @@ export const writeToCache = (content, delay = 500) => {
   }, delay);
 };
 
-export const removeDuplicatedModules = (url, cachedModules, moduleNames) => {
-  const matchingModule = moduleNames.find((moduleName) => url.match(new RegExp(`\\b\\/${moduleName}\\/\\b`)));
+const stripVersion = (url) => {
+  const parts = url.split('/');
+  parts.splice(-2, 1);
+  return parts.join('/');
+};
 
-  const updatedCachedModules = cachedModules;
-  Object.keys(updatedCachedModules).forEach((cachedModuleKey) => {
-    if (cachedModuleKey.match(new RegExp(`\\b\\/${matchingModule}\\/\\b`))) {
-      delete updatedCachedModules[cachedModuleKey];
-      console.log(`Deleted ${cachedModuleKey} from cache`);
-    }
-  });
+export const removeExistingEntryIfConflicting = (url, cachedModuleFiles) => {
+  const updatedCachedModules = cachedModuleFiles;
```

Comment:
delete will remove matching module from both `updatedCachedModules` and `cachedModuleFiles`. so the test in https://github.com/americanexpress/one-app/pull/1210/files#diff-0264af67b0d133bfc5c2b5d595e25c416dbc5e86ed199e8cd64d9924d99b2126L245 will always be true even if it removes files it should not. 
 
```suggestion
  const updatedCachedModules = { ...cachedModuleFiles };
```

Reply:
[8a673e6](https://github.com/americanexpress/one-app/pull/1210/commits/8a673e65b815cf25e7000186e92a184ee7ac22a1)

Addressed